src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -96,13 +134,18 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
         assert personToEdit != null;
 
         Name updatedName = editPersonDescriptor.getName().orElse(personToEdit.getName());
+        Nric updatedNric = editPersonDescriptor.getNric().orElse(personToEdit.getNric());
         Phone updatedPhone = editPersonDescriptor.getPhone().orElse(personToEdit.getPhone());
         Email updatedEmail = editPersonDescriptor.getEmail().orElse(personToEdit.getEmail());
         Address updatedAddress = editPersonDescriptor.getAddress().orElse(personToEdit.getAddress());
+        Appointment updatedAppointment = editPersonDescriptor.getAppointment()
+                .orElse(personToEdit.getAppointment().get());
```

Comment:
`getAppointment()` might fail here, as it could potentially return an empty `Optional`. You could wrap it like this instead:
```
    Appointment updatedAppointment = editPersonDescriptor.getAppointment()
        .orElse(personToEdit.getAppointment().orElse(null));
```